# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/03 20:18:05-08:00 rusty@rustcorp.com.au 
#   [PATCH] netfilter: Fix ip_conntrack_proto_sctp exit on sysctl fail
#   
#   On failure from register_sysctl_table, we return with exit 0.  Oops.  init and
#   fini should also be static.  nfsim found these.
#   
#   Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# net/ipv4/netfilter/ip_conntrack_proto_sctp.c
#   2005/01/03 15:49:16-08:00 rusty@rustcorp.com.au +3 -2
#   netfilter: Fix ip_conntrack_proto_sctp exit on sysctl fail
# 
diff -Nru a/net/ipv4/netfilter/ip_conntrack_proto_sctp.c b/net/ipv4/netfilter/ip_conntrack_proto_sctp.c
--- a/net/ipv4/netfilter/ip_conntrack_proto_sctp.c	2005-01-04 18:46:31 -08:00
+++ b/net/ipv4/netfilter/ip_conntrack_proto_sctp.c	2005-01-04 18:46:31 -08:00
@@ -609,7 +609,7 @@
 static struct ctl_table_header *ip_ct_sysctl_header;
 #endif
 
-int __init init(void)
+static int __init init(void)
 {
 	int ret;
 
@@ -622,6 +622,7 @@
 #ifdef CONFIG_SYSCTL
 	ip_ct_sysctl_header = register_sysctl_table(ip_ct_net_table, 0);
 	if (ip_ct_sysctl_header == NULL) {
+		ret = -ENOMEM;
 		printk("ip_conntrack_proto_sctp: can't register to sysctl.\n");
 		goto cleanup;
 	}
@@ -639,7 +640,7 @@
 	return ret;
 }
 
-void __exit fini(void)
+static void __exit fini(void)
 {
 	ip_conntrack_protocol_unregister(&ip_conntrack_protocol_sctp);
 #ifdef CONFIG_SYSCTL
